File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>res;
        unordered_map<string,vector<string>>ump;
        for(auto &it:strs){
            string temp=it;
            sort(temp.begin(),temp.end());
            ump[temp].push_back(it);
        }
        for(auto it:ump)
            res.push_back(it.second);
        return res;
        
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> result= new ArrayList<>();
        Map<String,List<String>> mp= new HashMap<>();
        for(int i=0;i<strs.length;i++) {
            char[] temp=strs[i].toCharArray();
            Arrays.sort(temp);
            String tempStr= new String(temp);
            List<String> tempArray=mp.getOrDefault(tempStr,new ArrayList<>());
            tempArray.add(strs[i]);
            mp.put(tempStr,tempArray);
        }
        mp.forEach((a,b)-> result.add(b));
        return result;
        
    }
}